### PR TITLE
OVMM: Teleport action and configs

### PR DESCRIPTION
**Pull request by yvsriram**
## Motivation and Context

- Action for teleporting to waypoint
- Following @jimmytyyang's BaseWaypointToVelocityController implementation
- Uses @jimmytyyang's changes for supporting non-circular robot bases 

## How Has This Been Tested
So far in interactive_play. Tested with a simple frontier exploration policy.

## Types of changes

- **\[Development\]** A pull request that add new features to the [habitat-lab](/habitat-lab) task and environment codebase. Development Pull Requests must be small (less that 500 lines of code change), have unit testing, very extensive documentation and examples. These are typically new tasks, environments, sensors, etc... The review process for these Pull Request is longer because these changes will be maintained by our core team of developers, so make sure your changes are easy to understand!

## Checklist

- [x] My code follows the code style of this project.
- [ ] I have updated the documentation if required.
- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes if required.

### CODE REVIEW

File: habitat-lab/habitat/tasks/rearrange/actions/actions.py [hunk 1]
```diff
@@ -461,227 +461,87 @@ def step(self, ee_pos, **kwargs):
...
+            np.array([xz[0], xz[1], 0.0]) for xz in self._navmesh_offset
+        ]
+        cur_pos = [trans.transform_point(xyz) for xyz in nav_pos_3d]
+        goal_pos = [target_trans.transform_point(xyz) for xyz in nav_pos_3d]
+
+        # For step filter of offset positions
+        end_pos = []
+        for i in range(num_check_cylinder):
+            pos = self._sim.step_filter(cur_pos[i], goal_pos[i])
+            # Sanitize the height
```
Comment (ykarmesh) [inline habitat-lab/habitat/tasks/rearrange/actions/actions.py:506]: What do you mean by "sanitize the height"?

File: habitat-lab/habitat/tasks/rearrange/actions/actions.py [hunk 2]
```diff
@@ -461,227 +461,87 @@ def step(self, ee_pos, **kwargs):
...
+            pos[1] = 0.0
+            cur_pos[i][1] = 0.0
+            goal_pos[i][1] = 0.0
+            end_pos.append(pos)
+
+        # Planar move distance clamped by NavMesh
+        move = []
+        for i in range(num_check_cylinder):
```
Comment (ykarmesh) [inline habitat-lab/habitat/tasks/rearrange/actions/actions.py:514]: This for loop could be combined with the previous one

File: habitat-lab/habitat/tasks/rearrange/actions/actions.py [hunk 3]
```diff
@@ -461,227 +461,87 @@ def step(self, ee_pos, **kwargs):
...
+            move.append((end_pos[i] - goal_pos[i]).length())
+
+        # For detection of linear or angualr velocities
```
Comment (ykarmesh) [inline habitat-lab/habitat/tasks/rearrange/actions/actions.py:517]:
```suggestion
        # For detection of linear or angular velocities
```

Comment (ykarmesh) [reply habitat-lab/habitat/tasks/rearrange/actions/actions.py:517]: Unclear what you mean by "For detection of linear or angular velocities"

File: habitat-lab/habitat/tasks/rearrange/actions/actions.py [hunk 4]
```diff
@@ -461,227 +461,86 @@ def step(self, ee_pos, **kwargs):
...
+        # There is a collision if the difference between the clamped NavMesh position and target position is too great for any point.
```
Comment (ykarmesh) [inline habitat-lab/habitat/tasks/rearrange/actions/actions.py:518]:
```suggestion
        # There is a collision if the distance between the clamped navmesh position and target position is greater than the self._collision_threshold.
```

File: habitat-lab/habitat/tasks/rearrange/actions/actions.py [hunk 5]
```diff
@@ -694,97 +553,92 @@ def _capture_robot_state(self):
...
-                # Don't allow the step, revert back.
-                self._set_robot_state(before_trans_state)
-                self.cur_robot.sim_obj.transformation = trans
+        # Check if there is a collision
+        _, new_target_trans = self.collision_check(trans, target_trans)
+        # Update the base
+        self.cur_robot.sim_obj.transformation = new_target_trans
         if self.cur_grasp_mgr.snap_idx is not None:
             # Holding onto an object, also kinematically update the object.
             # object.
```
Comment (ykarmesh) [inline habitat-lab/habitat/tasks/rearrange/actions/actions.py:581]:
```suggestion
```

File: habitat-lab/habitat/tasks/rearrange/actions/actions.py [hunk 6]
```diff
@@ -461,227 +461,86 @@ def step(self, ee_pos, **kwargs):
...
-        self.controller = Controller(
-            v_max=max_lin_speed,
-            w_max=max_ang_speed,
-            lin_gain=lin_speed_gain,
-            ang_gain=ang_speed_gain,
-            lin_error_tol=lin_tol,
-            ang_error_tol=ang_tol,
-            max_heading_ang=max_heading_ang,
-        )
+        self._allow_back = config.allow_back
```
Comment (ykarmesh) [inline habitat-lab/habitat/tasks/rearrange/actions/actions.py:473]: _allow_back is not being used

File: habitat-lab/habitat/tasks/rearrange/actions/actions.py (hunk 3 above)
Comment (yvsriram) [reply habitat-lab/habitat/tasks/rearrange/actions/actions.py]: Not sure. Didn't notice while copying from [here](https://github.com/facebookresearch/habitat-lab/blob/6f7c6b9c906a60b104a31ca3ad7051dcaed5e0f9/habitat-lab/habitat/tasks/rearrange/actions/actions.py#L549). Removed this now.

File: habitat-lab/habitat/config/default_structured_configs.py [hunk 7]
```diff
@@ -137,6 +137,14 @@ class BaseVelocityActionConfig(ActionConfig):
     ang_speed: float = 10.0
     allow_dyn_slide: bool = True
     allow_back: bool = True
+    collision_threshold: float = 1e-5
+    navmesh_offset: Optional[List[float]] = None
+    min_displacement: float = 0.05  # minimum displacement that is allowed
+    max_displacement_along_axis: float = 0.5  # maximum displacement
```
Comment (yvsriram) [inline habitat-lab/habitat/config/default_structured_configs.py:143]: @ykarmesh do these values look good - do you think they take into account the resolution of the controller?

Comment (ykarmesh) [reply habitat-lab/habitat/config/default_structured_configs.py:143]: It really depends on how the real controller behaves. Lets ask Chris.

**Review by ykarmesh — APPROVED** [review]

**Pull request merged**